File: module-1/09_Classes_Encapsulation/student-exercise/java/src/main/java/com/techelevator/Television.java
package com.techelevator;

public class Television {
	private boolean isOn = false;
	private int currentChannel = 3;
	private int currentVolume = 2;
	
	
	
	public boolean isOn() {
		return isOn;
	}

	public int getCurrentChannel() {
		return currentChannel;
	}

	public int getCurrentVolume() {
		return currentVolume;
	}

	public void turnOff() {
		isOn = false;
	}
	
    public void turnOn() {
    	isOn = true;
    	currentChannel = 3;
    	currentVolume = 2;
    }
    
    public void changeChannel(int newChannel) {
    	if (isOn && 3 <= currentChannel && currentChannel <= 18 && currentChannel != newChannel) {
    		currentChannel = newChannel;
    	}
    }
    
    public void channelUp() {
    	if (isOn) {
    		if (currentChannel != 18) {
    			currentChannel++;
    		}else {
    			currentChannel = 3;
    		}
    	}
    }
    
    public void channelDown() {
    	if (isOn) {
    		if (currentChannel != 3) {
    			currentChannel--;
    		}else {
    			currentChannel = 18;
    		}
    	}
    }
    
    public void raiseVolume() {
    	if (isOn && currentVolume < 10) {
    		currentVolume++;
    	}
    }
    
    public void lowerVolume() {
    	if (isOn && currentVolume > 0) {
    		currentVolume--;
    	}
    }
}

File: module-1/09_Classes_Encapsulation/exercise-final/dotnet/Individual.Exercises/Classes/Television.cs
namespace Individual.Exercises.Classes
{
    public class Television
    {
        /// <summary>
        /// Current on/off status of the tv
        /// </summary>
        public bool IsOn { get; private set; }


        /// <summary>
        /// Current selected channel number for the tv. Channels are 3 to 18.
        /// </summary>
        public int CurrentChannel { get; private set; } = 3;


        /// <summary>
        /// Current volume level for the tv. Volume level ranges from 0 to 10.
        /// </summary>
        public int CurrentVolume { get; private set; } = 2;


        /// <summary>
        /// Turns the tv on. Also resets the volume level to 2
        /// </summary>
        public void TurnOn()
        {
            IsOn = true;
            CurrentVolume = 2;
            CurrentChannel = 3;
        }


        /// <summary>
        /// Tuns the tv off.
        /// </summary>
        public void TurnOff()
        {
            IsOn = false;
        }


        /// <summary>
        /// Changes the tv channel, but only if it is on.
        /// </summary>
        /// <param name="newChannel">new channel to turn to</param>
        public void ChangeChannel(int newChannel)
        {
            if (IsOn && newChannel > 3 && newChannel < 18)
            {
                CurrentChannel = newChannel;
            }
        }


        /// <summary>
        /// Increases the Current Channel by 1, only if it is on.
        /// </summary>
        public void ChannelUp()
        {
            if (IsOn)
            {
                if (CurrentChannel == 18)
                {
                    CurrentChannel = 3;
                }
                else
                {
                    CurrentChannel++;
                }
            }
        }


        /// <summary>
        /// Increases the Current Channel by 1, only if it is on.
        /// </summary>
        public void ChannelDown()
        {
            if (IsOn)
            {
                if (CurrentChannel == 3)
                {
                    CurrentChannel = 18;
                }
                else
                {
                    CurrentChannel--;
                }
            }
        }


        /// <summary>
        /// Raises the volume. It won't raise it past 10
        /// </summary>
        public void RaiseVolume()
        {
            if (IsOn && CurrentVolume < 10)
            {
                CurrentVolume++;
            }
        }


        /// <summary>
        /// Lowers the volume. It won't lower it below 0
        /// </summary>
        public void LowerVolume()
        {
            if (IsOn && CurrentVolume > 0)
            {
                CurrentVolume--;
            }
        }
    }
}
